ReactiveCocoa/NSObject+Intercepting.swift

```diff
@@ -6,30 +6,419 @@ extension Reactive where Base: NSObject {
+	       "unknown method return type not supported in type encoding: \(String(cString: types))")
+
+	assert(types.pointee != Int8(UInt8(ascii: "(")), "union method return type not supported")
+	assert(types.pointee != Int8(UInt8(ascii: "{")), "struct method return type not supported")
+	assert(types.pointee != Int8(UInt8(ascii: "[")), "array method return type not supported")
+
+	assert(types.pointee != Int8(UInt8(ascii: "j")), "complex method return type not supported")
+
+	return true
+}
+
+/// Extract the arguments of an `NSInvocation` as an array of objects.
+///
+/// - parameters:
+///   - invocation: The `NSInvocation` to unpack.
+///
+/// - returns:
+///   An array of objects.
+private func unpackInvocation(_ invocation: AnyObject) -> [Any?] {
+	let invocation = invocation as AnyObject
+	let methodSignature = invocation.objcMethodSignature!
+	let count = UInt(methodSignature.numberOfArguments!)
+
+	var bridged = [Any?]()
+	bridged.reserveCapacity(Int(count - 2))
+
+	// Ignore `self` and `_cmd` at index 0 and 1.
+	for position in 2 ..< count {
+		let rawEncoding = methodSignature.argumentType(at: position)
+		let encoding = ObjCTypeEncoding(rawValue: rawEncoding.pointee) ?? .undefined
+
+		func extract<U>(_ type: U.Type) -> U {
+			let pointer = UnsafeMutableRawPointer.allocate(bytes: MemoryLayout<U>.size,
+			                                               alignedTo: MemoryLayout<U>.alignment)
+			defer {
+				pointer.deallocate(bytes: MemoryLayout<U>.size,
+				                   alignedTo: MemoryLayout<U>.alignment)
+			}
+
+			invocation.copy(to: pointer, forArgumentAt: Int(position))
+			return pointer.assumingMemoryBound(to: type).pointee
+		}
+
+		switch encoding {
+		case .char:
+			bridged.append(NSNumber(value: extract(CChar.self)))
+		case .int:
+			bridged.append(NSNumber(value: extract(CInt.self)))
+		case .short:
+			bridged.append(NSNumber(value: extract(CShort.self)))
+		case .long:
+			bridged.append(NSNumber(value: extract(CLong.self)))
+		case .longLong:
+			bridged.append(NSNumber(value: extract(CLongLong.self)))
+		case .unsignedChar:
+			bridged.append(NSNumber(value: extract(CUnsignedChar.self)))
+		case .unsignedInt:
+			bridged.append(NSNumber(value: extract(CUnsignedInt.self)))
+		case .unsignedShort:
+			bridged.append(NSNumber(value: extract(CUnsignedShort.self)))
+		case .unsignedLong:
+			bridged.append(NSNumber(value: extract(CUnsignedLong.self)))
+		case .unsignedLongLong:
+			bridged.append(NSNumber(value: extract(CUnsignedLongLong.self)))
+		case .float:
+			bridged.append(NSNumber(value: extract(CFloat.self)))
+		case .double:
+			bridged.append(NSNumber(value: extract(CDouble.self)))
+		case .bool:
+			bridged.append(NSNumber(value: extract(CBool.self)))
+		case .object:
+			bridged.append(extract((AnyObject?).self))
+		case .type:
+			bridged.append(extract((AnyClass?).self))
+		case .selector:
+			bridged.append(extract((Selector?).self))
```

Comment:
I'd move the `bridge.append` out of the `switch` so that the logic isn't repeated for every case. Then the compiler can also enforce that we set append one value in every case.

```swift
let value: Any?
switch ... {
    case .selector:
        value = extract((Selector?).self)
    ...
}
bridged.append(value)
```